<!--
  Cart presentation:
  - Update cart params
  - cart estimation
  - cart articulation
  - rviz view (optional)
 -->
<launch>
  <arg name="rviz" default="true"/>

  <node pkg="robot_state_publisher" type="state_publisher" name="cart_state_publisher" >
    <remap from="robot_description" to="cart_description" />
    <remap from="joint_states" to="cart_joint_states" />
  </node>

  <!-- upload cart description -->
  <rosparam file="$(find cob_pushing)/robots/cart/cart.urdf" command="load" ns="cart_description"/>

  <node pkg="cob_pushing" type="fake_cart_joint_state_publisher.py" name="cart_joint_state_publisher"/>

  <!--  **************** Visualisation ****************  -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find cob_pushing)/rviz/test_cart_urdf.rviz" if="$(arg rviz)"/>

</launch>
